link-grammar/preparation.c

```diff
@@ -30,8 +30,12 @@ set_connector_list_length_limit(Connector *c,
 {
 	for (; c!=NULL; c=c->next)
 	{
-		if (all_short ||
-		    (conset != NULL && !match_in_connector_set(conset, c)))
+		if (0 == strncmp(c->string, "ZZZ", 3))
+		{
+			c->length_limit = 1;
+		}
```

Comment:
according to my measurements; doing this actually gives a run that is a few percent slower than not doing this.  I just pushed a change which replaces it by string_set_compare, and only then is it a few percent faster... (for the In vivo sentence)